pkg/asset/agent/configimage/configiso.go

```diff
@@ -0,0 +1,160 @@
+	kubeadminPasswordHash       string = "/opt/agent/tls/kubeadmin-password.hash" //nolint:gosec // not hardcoded credentials
+	kubeconfigTLSCerts          string = "/opt/agent/tls/admin-kubeconfig-signer"
+	kubeapiserverTLSCerts       string = "/opt/agent/tls/kube-apiserver"
+)
+
+// TODO - need to add rendezvous host file when available.
+var fileFilters = []string{
+	pullSecretManifest,
+	nmStateConfigManifest,
+	clusterDeploymentManifest,
+	clusterImageSetManifest,
+	agentClusterInstallManifest,
+	extraManifests,
+	hostConfigFiles,
+	hostNameFiles,
+	kubeadminPasswordHash,
+	kubeconfigTLSCerts,
+	kubeapiserverTLSCerts,
+}
+
+// ConfigImage is an asset that generates a configuration ISO that can be used configure hosts.
+type ConfigImage struct {
+	tmpPath string
+}
+
+var _ asset.WritableAsset = (*ConfigImage)(nil)
+
+// Dependencies returns the assets on which the Config asset depends.
+func (a *ConfigImage) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&image.Ignition{},
+	}
+}
+
+// Generate generates the configuration image file.
+func (a *ConfigImage) Generate(dependencies asset.Parents) error {
+	ignition := &image.Ignition{}
+
+	dependencies.Get(ignition)
+
+	ca := image.NewCpioArchive()
+
+	// Walk the list of ignition files.
+	// For the ones to be included in the configimage add to cpio archive
+	for _, file := range ignition.Config.Storage.Files {
+		filename := file.Node.Path
+		mode := *file.FileEmbedded1.Mode
+		if includeFileInImage(filename) {
+			logrus.Debugf("Including file %s", filename)
+			contents := ""
+			if file.FileEmbedded1.Contents.Source != nil {
+				output, err := dataurl.DecodeString(*file.FileEmbedded1.Contents.Source)
+				if err != nil {
+					return errors.Wrapf(err, "decode string failed for %s", filename)
+				}
+				contents = string(output.Data)
+			}
+
+			err := ca.StoreBytes(filename, []byte(contents), mode)
+			if err != nil {
+				return errors.Wrapf(err, "failure storing file %s in cpio", filename)
+			}
+		}
+	}
+
+	// Save the cpi archive to a temp file
+	tmpPath, err := os.MkdirTemp("", "agent")
+	if err != nil {
+		return err
+	}
+	a.tmpPath = tmpPath
+
+	output := filepath.Join(tmpPath, configCpioArchive)
+
+	err = ca.Save(output)
+	if err != nil {
+		return errors.Wrapf(err, "failure saving cpio file %s", configCpioArchive)
+	}
+
+	return nil
+}
+
+// PersistToFile writes the iso image in the assets folder.
+func (a *ConfigImage) PersistToFile(directory string) error {
+	defer os.RemoveAll(a.tmpPath)
+
+	// If the tmpPath are not set then it means that either one of the Config
+	// dependencies or the asset itself failed for some reason
+	if a.tmpPath == "" {
+		return errors.New("cannot generate Config image due to configuration errors")
+	}
+
+	configImageFile := filepath.Join(directory, configImageFilename)
+
+	// Remove symlink if it exists
+	os.Remove(configImageFile)
+
+	logrus.Infof("In PersistToFile, creating %s", configImageFile)
```

Comment:
Let's remove the name of the internal function `PersistToFile` being logged here

Reply:
Will remove